### ISSUE TITLE
Comet boots without functioning network, and missing system utilities.

### ISSUE THREAD

**Issue by nrhtr** [urbit/urbit#2000]
Hi,

I've tried to boot a fresh comet on multiple machines (2 macOS, 1 Debian) and while I get a dojo> prompt, it seems something has gone wrong in the boot process as I get no response to `|hi ~zod` or `|hi ~marzod` and normal commands like `+trouble` give me a "scry" error.

Here's a full log of everything I did.

Downloaded latest macOS binary from https://urbit.org/using/install/ and compared to the version I have.

```
~/newurbit shasum -a 256 ./urbit
7fd31d30287a8e53a3486d43432fc4386b99b8347dd01f413cd273e11abd72e5  ./urbit
~/newurbit shasum -a 256 ../urbit
7fd31d30287a8e53a3486d43432fc4386b99b8347dd01f413cd273e11abd72e5  ../urbit
```

Deleted old comet and booted a new one
```
~ rm -rf robcomet
~ ./urbit -c freshcomet
~
urbit 0.9.2
boot: home is /Users/jparker/freshcomet
loom: mapped 2048MB
lite: arvo formula 52fd6bb6
lite: core 6ce08503
lite: final state 6ce08503
boot: downloading pill https://bootstrap.urbit.org/urbit-v0.9.2.pill
boot: mining a comet. May take up to an hour.
If you want to boot faster, get an Azimuth point.
boot: found comet ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod
boot: retrieving latest block
boot: verifying keys
boot: getting sponsor
boot: retrieving galaxy table
boot: retrieving network domains
boot: retrieving keys for sponsor ~marzod
boot: retrieving keys for sponsor ~zod
loom: mapped 2048MB
boot: protected loom
live: logical boot
boot: installed 262 jets
work: play 1
boot: ship: ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod
work: (1)| boot
work: (2)| boot
work: (3)| boot
work: (4)| boot
work: (5)| boot
work: (5)| pill: d1f7e9a
1-b
1-c (compiling compiler, wait a few minutes)
ride-parsing
ride-compiling
ride-compiled
1-d
1-e
ride-parsing
ride-compiling
ride-compiled
1-f
ride-parsing
ride-compiling
ride-compiled
1-g
%arvo-assembly
%arvo-assembled
work: (5)| core: 749aed72
[%tang /~zod/home/~2019.11.14..13.38.55..0ecd/sys/zuse ~difdel-davdev]
arvo: metamorphosis
%hoon-load
[%vane %a /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/ames ~dindep-mirzod]
[%vane-parsed ~ponnyx-bondyn]
[%vane-compiled ~barmud-silsed]
[%vane %b /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/behn ~pittyp-milsug]
[%vane-parsed ~fontud-rovref]
[%vane-compiled ~daclur-minlud]
[%vane %c /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/clay ~padmyr-fipsup]
[%vane-parsed ~minfyn-modbep]
[%vane-compiled ~hidwes-mochul]
[%vane %d /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/dill ~narlun-navfyl]
[%vane-parsed ~marleg-lodsep]
[%vane-compiled ~ticnec-sovdyt]
[%vane %e /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/eyre ~watred-siptex]
[%vane-parsed ~torhul-rovmes]
[%vane-compiled ~hassyp-malsev]
[%vane %f /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/ford ~bindeg-torsed]
[%vane-parsed ~bannut-tommur]
[%vane-compiled ~natmul-masfel]
[%vane %g /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/gall ~ronlyn-napteb]
[%vane-parsed ~habwyd-tilled]
[%vane-compiled ~ravned-ravlev]
[%vane %i /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/iris ~socmel-magwyn]
[%vane-parsed ~tobrec-losweg]
[%vane-compiled ~tinpem-picrud]
[%vane %j /~zod/home/~2019.11.14..13.38.55..0ecd/sys/vane/jael ~tidber-simmet]
[%vane-parsed ~togmul-nibpyx]
[%vane-compiled ~tarnul-mocdex]
pier: boot complete
>>  [%gall-not-running %azimuth-tracker %poke]
>>  [%gall-not-running %azimuth-tracker %peer]
clay: committed initial filesystem (hoon)
http: live (insecure, public) on 80
http: live (insecure, loopback) on 12321
clay: committed initial filesystem (all)
finding ship and desk from %base on ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod to %home
>>  [%gall-not-running %dojo %peer]
>>  [%gall-not-running %chat-cli %peer]
>>  [%gall-not-running %launch %poke]
>>  [%gall-not-running %launch %poke]
>>  [%gall-not-running %invite-store %peer]
>>  [%gall-not-running %chat-store %peer]
[~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod, driving ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod]
activated app home/dojo
activated app home/chat-cli
activated app home/chat-hook
activated app home/chat-store
activated app home/chat-view
activated app home/clock
activated app home/group-hook
activated app home/group-store
activated app home/invite-hook
activated app home/invite-store
activated app home/invite-view
activated app home/launch
activated app home/lens
activated app home/modulo
activated app home/permission-group-hook
activated app home/permission-hook
activated app home/permission-store
activated app home/publish
activated app home/weather
[linked to [p=~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod q=%chat-cli]]
[linked to [p=~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod q=%dojo]]
activated sync from %base on ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod to %home
sync succeeded from %base on ~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod to %home
~nosset_marzod:dojo>

> (sein:title our now our)
~marzod
ames: czar zod.urbit.org: ip .34.83.15.224
; ~marzod not responding still trying
; ~zod not responding still trying
> .^(* %j /=life=/~zod)
ford: %ride: invalid scry path: /j/~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod/life/~2019.11.28..02.40.35..8e04/~zod
~nosset_marzod:dojo>
```

Edit: also here's the output from tying to run `+trouble`
```
> +trouble
ford: %call: invalid scry path: /j/~nosset-wolsyr-lasmud-novtux--ladhex-morweb-libsyn-marzod/life/~2019.11.28..04.40.33..9c4a/~marzod
```

**Comment by matildepark** [comment]
I think comets have a key propagation issue and they don’t get OTAs, so +trouble wouldn’t help you.

There were instructions from ~palfun earlier today but I can’t recall now. I believe you want:

|start %azimuth-tracker
:azimuth-tracker|listen ~ %app %azimuth-tracker
|sync %base (sein:title our now our) %kids

—
~haddef-sigwen
https://urbit.org

**Comment by nrhtr** [comment]
Aha, thanks! That seems to have fixed it. I did try those commands before and it didn't seem to work. I might have missed one or done something else incorrectly. Still it would be nice if comets would "just work". Is there another issue tracking that or would this be one?

Thanks again.

**Comment by matildepark** [comment]
Presumably they will once the next patch lands?

**Comment by nrhtr** [comment]
Oh right. I see what you mean now. So it's a case of the latest OTA not being included in the download/boot process?

Although, wouldn't the same issue occur after the next OTA?

**Comment by jfranklin9000** [comment]
`+trouble` should definitely fail more gracefully when it doesn't yet have keys for other ships.  I plan to convert `+trouble`'s output to sole and I will take a look at this when I do that.

How would I catch the scry error?

**Comment by philipcmonk** [comment]
I'm not sure exactly how to virtualize a .^ call.  The easiest solution might be to add a path in jael to check if we know anything about a ship and produce & or |, so you can check that before trying the /=life= path.  You could do this by looking in sys/vane/jael.hoon for ++  scry, then copy the %life case, except change it to never produce ~, instead producing [~ ~ !>([%noun %|])], and for the cases where it produces an [%atom ...], change to [%noun %&].
